indexer/lib/proof-set-handler.js

```diff
@@ -20,16 +20,25 @@ export async function handleProofSetRailCreated(
     CHAINALYSIS_API_KEY,
   } = env
 
-  let isPayerSanctioned
-  try {
-    if (payload.with_cdn) {
-      isPayerSanctioned = await isAddressSanctioned(payload.payer, {
+  if (payload.with_cdn) {
+    try {
+      const isPayerSanctioned = await isAddressSanctioned(payload.payer, {
         CHAINALYSIS_API_KEY,
       })
+
+      await env.DB.prepare(
+        `
+        INSERT INTO wallet_details (wallet_address, is_sanctioned)
+        VALUES (?, ?)
+        ON CONFLICT (wallet_address) DO UPDATE SET is_sanctioned = excluded.is_sanctioned
+      `,
+      )
+        .bind(payload.payer, isPayerSanctioned)
+        .run()
+    } catch (err) {
+      console.error(`Error checking if payer is sanctioned: ${err}`)
+      throw err // Let caller handle the error
```

Comment:
If we are just re-throwing the error to let the caller handle it, do we need the try/catch block at all?